backend/src/GraphQl/OpenIdConnect/Application/ApplicationQueries.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using HotChocolate.Types;
+using Metabase.Authorization;
+using Metabase.Data;
+using Metabase.GraphQl.Users;
+using Microsoft.AspNetCore.Identity;
+using OpenIddict.Core;
+using OpenIddict.EntityFrameworkCore.Models;
+
+namespace Metabase.GraphQl.OpenIdConnect.Application;
+
+[ExtendObjectType(nameof(Query))]
+public sealed class ApplicationQueries
+{
+    // TODO In all queries, instead of returning nothing, report as authentication error to client.
+    [UseUserManager]
+    public async Task<IList<OpenIddictEntityFrameworkCoreApplication>> GetApplications(
+        OpenIddictApplicationManager<OpenIddictEntityFrameworkCoreApplication> applicationManager,
+        ClaimsPrincipal claimsPrincipal,
+        UserManager<User> userManager,
+        Data.ApplicationDbContext context, // TODO Make the application manager use the scoped database context.
+        CancellationToken cancellationToken
+    )
+    {
+        if (!await OpenIdConnectAuthorization.IsAuthorizedToViewApplications(claimsPrincipal, userManager, context, cancellationToken)
+                .ConfigureAwait(false))
+        {
+            return Array.Empty<OpenIddictEntityFrameworkCoreApplication>();
```

Comment:
I would prefer to turn that into a proper error instead of just returning an empty array. Otherwise it is unclear to the user if he/she has no applications to manage or doesn't have the rights to do so (maybe because he/she is not logged in or because somebody retracted rights or ...). I don't know how to accomplish that though in queries. One possibility would be to use payloads as in mutations. That seems to be too much overhead though to me. How do other well-known GraphQL APIs do that? For example the one from GitHub. We could for example return an empty array as we already do and report an error via GraphQL (GraphQL allows returning data and errors in the same response). I'll make some investigations regarding that issue @simon-wacker